include/c2pa.h

```diff
@@ -135,15 +213,284 @@ char *c2pa_sign_file(const char *source_path,
+
+/**
+ * writes a C2paReader resource to a stream given a uri
+ * # Errors
+ * Returns -1 if there were errors, otherwise returns size of stream written
+ *
+ * # Safety
+ * Reads from null terminated C strings
+ *
+ * # Example
+ * ```c
+ * result c2pa_reader_resource_to_stream(store, "uri", stream);
+ * if (result < 0) {
+ *    printf("Error: %s\n", c2pa_error());
+ * }
+ * ```
+ */
+IMPORT extern
+int c2pa_reader_resource_to_stream(struct C2paReader *reader_ptr,
+                                   const char *uri,
+                                   struct CStream *stream);
+
+/**
+ * Creates a C2paBuilder from a JSON manifest definition string
+ * # Errors
+ * Returns NULL if there were errors, otherwise returns a pointer to a Builder
+ * The error string can be retrieved by calling c2pa_error
+ * # Safety
+ * Reads from null terminated C strings
+ * The returned value MUST be released by calling c2pa_builder_free
+ * and it is no longer valid after that call.
+ * # Example
+ * ```c
+ * auto result = c2pa_builder_from_json(manifest_json);
+ * if (result == NULL) {
+ *  printf("Error: %s\n", c2pa_error());
+ * }
+ * ```
+ *
+ */
+IMPORT extern struct C2paBuilder *c2pa_builder_from_json(const char *manifest_json);
+
+/**
+ * Create a C2paBuilder from an archive stream
+ * # Errors
+ * Returns NULL if there were errors, otherwise returns a pointer to a Builder
+ * The error string can be retrieved by calling c2pa_error
+ * # Safety
+ * Reads from null terminated C strings
+ * The returned value MUST be released by calling c2pa_builder_free
+ * and it is no longer valid after that call.
+ * # Example
+ * ```c
+ * auto result = c2pa_builder_from_archive(stream);
+ * if (result == NULL) {
+ * printf("Error: %s\n", c2pa_error());
+ * }
+ * ```
+ */
+IMPORT extern struct C2paBuilder *c2pa_builder_from_archive(struct CStream *stream);
+
+/**
+ * Frees a C2paBuilder allocated by Rust
+ * # Safety
+ * can only be freed once and is invalid after this call
+ */
+IMPORT extern void c2pa_builder_free(struct C2paBuilder *builder_ptr);
+
+/**
+ * Adds a resource to the C2paBuilder
+ * # Errors
+ * Returns -1 if there were errors, otherwise returns 0
+ * The error string can be retrieved by calling c2pa_error
+ * # Safety
+ * Reads from null terminated C strings
+ *
+ */
+IMPORT extern
+int c2pa_builder_add_resource(struct C2paBuilder *builder_ptr,
+                              const char *uri,
+                              struct CStream *stream);
+
+/**
+ * Adds an ingredient to the C2paBuilder
+ *
+ * # Parameters
+ * * builder_ptr: pointer to a Builder
+ * * ingredient_json: pointer to a C string with the JSON ingredient definition
+ * * format: pointer to a C string with the mime type or extension
+ * * source: pointer to a CStream
+ * # Errors
+ * Returns -1 if there were errors, otherwise returns 0
+ * The error string can be retrieved by calling c2pa_error
+ * # Safety
+ * Reads from null terminated C strings
+ *
+ */
+IMPORT extern
```

Comment:
Choose a consistent style for `IMPORT extern` on same line or not?

Reply:
Still inconsistent.

Reply:
This file is auto generated from the rust code via cbindgen, so I"m nos sure if I have control over this. I'll take a look.